### PR TITLE
[CI] Build & Test Fabric iOS, Disable failing CircleCI checks

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

- Add a new job to build Fabric iOS on Azure Pipelines. While here, let's also rename the jobs to better reflect what they do
- Make some Fabric code fixes to get the new Fabric iOS job passing
- Disable the CircleCI tests that are failing while we are mid 0.71-merge

End result: Green CI checks 🎉

## Changelog

[Type] [Added] - Build Fabric iOS on Azure Pipelines

## Test Plan

CI passes

### CODE REVIEW

**Comment by christophpurrer** [comment]
To fix the current linking issue for
```
2023-01-14T08:17:33.4525870Z     Undefined symbol: _RNTMyNativeViewCls
2023-01-14T08:17:33.4526550Z     Testing cancelled because the build failed.
```

you should merge in https://github.com/microsoft/react-native-macos/pull/1520 first from @shwanton

File: packages/rn-tester/RNTester/NativeExampleViews/UpdatePropertiesExampleView.m [hunk 1]
```diff
@@ -53,16 +53,18 @@ - (instancetype)initWithFrame:(CGRect)frame
                                          moduleName:@"SetPropertiesExampleApp"
                                   initialProperties:@{@"color" : @"beige"}];
 
-#if !TARGET_OS_OSX // [macOS]
+    // [macOS Github#1642: Suppress analyzer error of nonlocalized string
+    NSString *buttonTitle = NSLocalizedString(@"Native Button", nil);
+#if !TARGET_OS_OSX
     _button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
-    [_button setTitle:@"Native Button" forState:UIControlStateNormal];
+    [_button setTitle:buttonTitle forState:UIControlStateNormal];
```
Comment (mischreiber) [inline packages/rn-tester/RNTester/NativeExampleViews/UpdatePropertiesExampleView.m:60]: Possible inline comment to help with future merges?

Comment (Saadnajmi) [reply packages/rn-tester/RNTester/NativeExampleViews/UpdatePropertiesExampleView.m]: Added!

**Comment by Saadnajmi** [comment]
> To fix the current linking issue for
> 
> ```
> 2023-01-14T08:17:33.4525870Z     Undefined symbol: _RNTMyNativeViewCls
> 2023-01-14T08:17:33.4526550Z     Testing cancelled because the build failed.
> ```
> 
> you should merge in #1520 first from @shwanton

I still get this error if `RCT_NEW_ARCH_ENABLED` is set. I'm going to chalk it up to "At this point in time in our repo, the new architecture / codegen are still experimental", and that flag enables more than fabric. I've decided to not set that flag to avoid that error.

**Comment by christophpurrer** [comment]
> RCT_NEW_ARCH_ENABLED

I think for rn 0.68 you need to set `USE_FABRIC` ...
The flag `RCT_NEW_ARCH_ENABLED` wasn't used for Fabric afaik

**Review by mischreiber — APPROVED** [review]

**Pull request merged**